### PR TITLE
FIX tc deletion inside insertion [MACRO-1723]

### PR DESCRIPTION
**Pull request by synoet**
# Summary of PR
When you create a deletion redline inside the range of an existing insertion redline, the initial redline gets split up.

"Initial Test Redline"(Insertion)
---- *DELETE* "Test" ----
"Initial"(Insertion), "Test"(Deletion), "Redline"(Insertion) 

The existing code would push the RedlineData from the first "Initial" redline to the split "Test" redline.
But this is incorrect, because the "Test" redline has different author information.

These changes should be isolated exclusively to the case where you make a deletion within an existing insertion.

The inverse of this, making an insertion within an existing deletion currently works correctly, and it does not push the data to the split parts.

```cpp
case RedlineType::Insert:
    switch (pRedl->GetType())
    {
        // ... other cases ...

        case RedlineType::Delete:
            if (SwComparePosition::Inside == eCmpPos)
            {
                // split up
                if (*pEnd != *pREnd)
                {
                    SwRangeRedline* pCpy = new SwRangeRedline(*pRedl);
                    pCpy->SetStart(*pEnd);
                    maRedlineTable.Insert(pCpy);
                }
                pRedl->SetEnd(*pStt, pREnd);
                if ((*pStt == *pRStt) && (pRedl->GetContentIdx() == nullptr))
                {
                    maRedlineTable.DeleteAndDestroy(n);
                    bDec = true;
                }
                else if (!pRedl->HasValidRange())
                {
                    // re-insert
                    maRedlineTable.Remove(n);
                    maRedlineTable.Insert(pRedl);
                }
            }
            break;
    }
    break;

```

## Before you submit this PR
This is a **public, open source project**. Confidential or sensitive information should not be included in this description or any comments, including but not limited to links, files, and documents.

*If you're not certain that information is not confidential or sensitive in nature, do not include it.*

- [ ] I have verified that this PR does not include any confidential information

### CODE REVIEW

**Review by chase — COMMENTED** [review]
This fix seems familiar, and I'm worried it will negatively affect the rest of track changes because the push was added but the corresponding pop aren't removed.

This seems like it might be an issue upstream, can you verify @synoet?

Might be a good idea to consult with Collabora about this fix.

**Comment by synoet** [comment]
@chase I might be wrong but I think there is no need for a correspond pop since we where calling it with `bOwnAsNext = false`

Which I believe would mean that there isn't actually anything that would need to be popped since we aren't directly changing just appending? 

I did do some general test... Though you are right no harm in consulting collabora.
```cpp
void SwRangeRedline::PushData( const SwRangeRedline& rRedl, bool bOwnAsNext )
{
    SwRedlineData* pNew = new SwRedlineData( *rRedl.m_pRedlineData, false );
    if( bOwnAsNext )
    {
        pNew->m_pNext = m_pRedlineData;
        m_pRedlineData = pNew;
    }
    else
    {
        pNew->m_pNext = m_pRedlineData->m_pNext;
        m_pRedlineData->m_pNext = pNew;
    }
}
```

**Comment by chase** [comment]
> @chase I might be wrong but I think there is no need for a correspond pop since we where calling it with `bOwnAsNext = false`

You very well could be right, but I think consulting with Collabora on this would be best, don't want to introduce regressions.

File: libreoffice-core/sw/source/core/doc/DocumentRedlineManager.cxx [hunk 1]
```diff
@@ -1921,13 +1921,22 @@ DocumentRedlineManager::AppendRedline(SwRangeRedline* pNewRedl, bool const bCall
...
                             if( *pRStt == *pStt )
                             {
                                 // #i97421#
                                 // redline w/out extent loops
+                                // new redline has some content
                                 if (*pStt != *pEnd)
                                 {
-                                    pNewRedl->PushData( *pRedl, false );
+
+                                    // MACRO : {
```
Comment (chase) [inline libreoffice-core/sw/source/core/doc/DocumentRedlineManager.cxx:1933]:
```suggestion
                                    // MACRO-1723: Fix author when deleting an existing insertion {
```

Please add/fix the remaining code for the rest of the changes

**Review by chase — CHANGES_REQUESTED** [review]
Overall good, just need to add the MACRO comments

File: libreoffice-core/sw/inc/redline.hxx
Comment (chase) [inline libreoffice-core/sw/inc/redline.hxx:1]: Please add comments here as well

File: libreoffice-core/sw/source/core/doc/docredln.cxx
Comment (chase) [inline libreoffice-core/sw/source/core/doc/docredln.cxx:1]: Please add comments here as well

**Review by chase — CHANGES_REQUESTED** [review]
All proprietary changes need comments

**Pull request merged**